configure.ac

```diff
@@ -3763,6 +3763,28 @@ dnl The AIX_BUILDDATE is obtained from the kernel fileset - bos.mp64
 	*) ;;
 esac
 
+# check for _Complex C type
+#
+# Note that despite most compilers define __STDC_IEC_559_COMPLEX__ - almost
+# none properly support C11+ Annex G (where pure imaginary types
+# represented by _Imaginary are mandatory).  This is a bug (see e.g.
+# llvm/llvm-project#60269), so we don't rely on presence
+# of __STDC_IEC_559_COMPLEX__.
+AC_RUN_IFELSE([AC_LANG_SOURCE([[
+#include <complex.h>
+int main(void)
+{
+    double complex z = 1 + 2*I;
+    z = z + z*I;
+    return (creal(z) != -1) || (cimag(z) != 3);
+}]])], [ac_cv_c_complex_supported=yes],
+[ac_cv_c_complex_supported=no],
+[ac_cv_c_complex_supported=no])
+if test "$ac_cv_c_complex_supported" = "yes"; then
+    AC_DEFINE([HAVE_C_COMPLEX], [1],
```

Comment:
Would it be possible to add PY_ prefix to the newly added HAVE_ macro? Also, it doesn't seem like you're testing "_Complex" but "double complex". So what do you think of renaming the macro to PY_HAVE_DOUBLE_COMPLEX?

Reply:
> Would it be possible to add PY_ prefix to the newly added HAVE_ macro?

Sure, but this seems a bit unusual for configure-born macros (I see no such examples in ``pyconfig.h``).

> Also, it doesn't seem like you're testing "_Complex" but "double complex".

The "complex" is just an alias for ``_Complex`` type specifier.  Standard says: "There are three complex types, designated as float _Complex, double _Complex, and long double _Complex."  In future we might want to add (see pr description) also support for "float complex" and "long double complex".

I'll add tests for "float" and "long double" right now.

Reply:
@erlend-aasland and @encukou once proposed to rename HAVE_ constant to PY_HAVE_ constants to not pollute the Python.h API with names not starting with Py or PY. IMO we should consider starting doing that for new macros.

Reply:
That would be `Py_HAVE`, _not_ `PY_HAVE`: https://peps.python.org/pep-0007/#naming-conventions

Reply:
There are some:

```sh
$  git grep "undef P[yY]_" pyconfig.h.in                                                                                                                                                (main %)
pyconfig.h.in:#undef PY_BUILTIN_HASHLIB_HASHES
pyconfig.h.in:#undef PY_COERCE_C_LOCALE
pyconfig.h.in:#undef PY_HAVE_PERF_TRAMPOLINE
pyconfig.h.in:#undef PY_SQLITE_ENABLE_LOAD_EXTENSION
pyconfig.h.in:#undef PY_SQLITE_HAVE_SERIALIZE
pyconfig.h.in:#undef PY_SSL_DEFAULT_CIPHERS
pyconfig.h.in:#undef PY_SSL_DEFAULT_CIPHER_STRING
pyconfig.h.in:#undef PY_SUPPORT_TIER
pyconfig.h.in:#undef Py_DEBUG
pyconfig.h.in:#undef Py_ENABLE_SHARED
pyconfig.h.in:#undef Py_GIL_DISABLED
pyconfig.h.in:#undef Py_HASH_ALGORITHM
pyconfig.h.in:#undef Py_RL_STARTUP_HOOK_TAKES_ARGS
pyconfig.h.in:#undef Py_STATS
pyconfig.h.in:#undef Py_SUNOS_VERSION
pyconfig.h.in:#undef Py_TRACE_REFS
```

Reply:
See also:

- https://github.com/python/cpython/pull/115343

Reply:
Renamed to Py_HAVE_C_COMPLEX

Reply:
Thanks; for new constants, let's try to follow the PEP-7 convention. We don't need to create more problems in pyconfig.h.in :)